servicebox-app/src/main/java/io/halkyon/resource/page/ClaimResource.java

```diff
@@ -68,12 +68,17 @@ private TemplateInstance showList(List<Claim> claims) {
     @Consumes(MediaType.APPLICATION_JSON)
     @Path("/name/{name}")
     public io.halkyon.model.Claim findByName(@PathParam("name") String name) {
-        return io.halkyon.model.Claim.findByName(name);
+        Claim claim = Claim.findByName(name);
+        if (claim == null) {
+            throw new WebApplicationException("Claim with name " + name + " does not exist.", 404);
```

Comment:
we should throw the NotFoundException exception instead.

Reply:
Take care that throwing an exception will return a HTTP 404 exception that it must be handled by HTMX. Ideally if there is such an error, we should like when we submit an error which is raising an error that a message is displayed in a box in red within the HTML page. See what we did for HTML Claim form by example